backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/SparkPlanExecApiImpl.scala

```diff
@@ -659,4 +662,81 @@ class SparkPlanExecApiImpl extends SparkPlanExecApi {
   ): GenerateExecTransformerBase = {
     GenerateExecTransformer(generator, requiredChildOutput, outer, generatorOutput, child)
   }
+
+  override def genPreProjectForGenerate(generate: GenerateExec): SparkPlan = {
+    if (supportsGenerate(generate)) {
+      val newGeneratorChild = generate.generator.asInstanceOf[UnaryExpression].child match {
+        case attr: Attribute => attr
+        case expr @ (Literal(_, _) | CreateMap(_, _) | CreateArray(_, _)) =>
+          Alias(expr, generatePreAliasName)()
+        case other =>
+          throw new UnsupportedOperationException(
+            s"Fail to execute ${generate.generator.getClass.getSimpleName} " +
+              s"with child type ${other.getClass.getSimpleName}")
+      }
+      generate.copy(
+        generator =
+          generate.generator.withNewChildren(Seq(newGeneratorChild)).asInstanceOf[Generator],
+        child = ProjectExec(generate.child.output :+ newGeneratorChild, generate.child)
```

Comment:
Use `eliminateProjectList` to get the `projectList` of pre project.

Reply:
Do you mean by using `elimainateProjectList(generate.child.output, newGeneratorChild)`? `newGeneratorChild` can be a duplicated Attribute in `generate.child.output`. It shouldn't be eliminated because the native backend identifies the last field as generator's input.

Reply:
Could you add some comments for this explanation?